rules-reviewed/camel3/camel2/component-changes.windup.groovy

```diff
@@ -0,0 +1,308 @@
+        )
+        .perform(Iteration.over("default").perform(
+                createHint("Terser language: language renamed",
+                        "`terser` language renamed to `hl7terser`",
+                        "languages", true)
+                        .withQuickfix(createReplaceQuickfix("org.apache.camel.component.hl7.HL7.terser",
+                                "org.apache.camel.component.hl7.HL7.hl17terser"))))
+        .withId("component-changes-00008")
+
+        .addRule()
+        .when(Or.any(JavaClass.references("org.apache.camel.converter.crypto.CryptoDataFormat{*}").at(TypeReferenceLocation.VARIABLE_DECLARATION),
+                XmlFile.matchesXpath("//*[c:crypto and not(b:crypto/@algorithm)]/c:crypto").namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*[b:crypto and not(b:crypto/@algorithm)]/b:crypto").namespace("b", "http://camel.apache.org/schema/blueprint"))
+        )
+        .perform(createHint("Crypto dataformat: The default encryption algorithm changed to null",
+                "The default encryption algorithm is mandatory changed from `DES/CBC/PKCS5Padding` to null. " +
+                        "Therefore if algorithm hasn't been set yet, it's required to set a value for it",
+                "crypto_dataformat", false))
+        .withId("component-changes-00009")
+
+        .addRule()
+        .when(Or.any(
+                XmlFile.matchesXpath("//*[c:secureXML and (count(c:secureXML/@passPhrase)+count(c:secureXML/@passPhraseByte))<1]/c:secureXML/")
+                        .namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*[b:secureXML and (count(b:secureXML/@passPhrase)+count(b:secureXML/@passPhraseByte))<1]/b:secureXML/")
+                        .namespace("b", "http://camel.apache.org/schema/blueprint"),
+                FileContent.matches("{*}secureXML").inFileNamed("{*}.java"))
+        )
+        .perform(createHint("XMLsecure dataformat: The default encryption key has been removed",
+                "The default encryption key has been removed, so it is now mandatory to supply " +
+                        "the key String/bytes if you are using symmetric encryption.",
+                "xml_security_dataformat", false))
+        .withId("component-changes-00010")
+
+        .addRule()
+        .when(Or.any(
+                XmlFile.matchesXpath("//*/c:from[@uri=windup:matches(self::node(), '{*}consumer.{*}')]")
+                        .namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*/b:from[@uri=windup:matches(self::node(), '{*}consumer.{*}')]")
+                        .namespace("b", "http://camel.apache.org/schema/blueprint"),
+                FileContent.matches("{*}from({*}consumer.{*})").inFileNamed("{*}.java"))
+        )
+        .perform(createHint("Consumer endpoints: options with consumer. prefix have been removed. ",
```

Comment:
No documentation link here ?

Reply:
Link is generated in `createHint(....)` method here: https://github.com/windup/windup-rulesets/blob/09f558be161a12b01c91fb5949f98523a0dfc53b/rules-reviewed/camel3/camel2/component-changes.windup.groovy#L50